backend/src/lang/intents/strategy/strategy.intent.ts

```diff
@@ -0,0 +1,39 @@
+import { IIntentStrategy } from './strategy.inteface';
+import { IIntentFactory } from '../factory/factory.interface';
+import { IntentHandler } from '../handlers/handler.abstract';
+import { Injectable } from '@nestjs/common';
+import { EndDateFactory } from '../factory/factory.enddate';
+import { AddStartDateFactory } from '../factory/factory.addstartdate';
+import { ContextFactory } from '../factory/factory.context';
+import { HelpFactory } from '../factory/factory.help';
+import { CreateContractFactory } from '../factory/factory.createcontract';
+import { EndDateOpenFactory } from '../factory/factory.enddateopen';
+
+@Injectable()
+export class IntentStrategy implements IIntentStrategy {
+    private readonly intentFactories: Array<IIntentFactory>;
+
+    constructor(private endDateFactory: EndDateFactory,
+                private addStartDateFactory: AddStartDateFactory,
+                private contextFactory: ContextFactory,
+                private helpFactory: HelpFactory,
+                private createContactFactory: CreateContractFactory,
+                private endDateOpenFactory: EndDateOpenFactory) {
+        this.intentFactories = [endDateFactory, addStartDateFactory, contextFactory, helpFactory, createContactFactory, endDateOpenFactory];
+    }
+
+    public createIntentHandler(action: string): IntentHandler{
+        let intentFactory: IIntentFactory;
+        this.intentFactories.forEach( (factory) => {
```

Comment:
Ginge auch kürzer:
`const intentFactory = this.intentFactories.find((factory) => factory.appliesTo(action)`

Was passiert, wenn mehere Factories matchen?